### PR TITLE
Handling NaN in consistent with Nodejs

### PR DESCRIPTION
**Pull request by geastwood**
This [bug](https://github.com/feross/buffer/issues/197) eventually leads to this PR.

### CODE REVIEW

**Comment by geastwood** [comment]
@feross would you take a look?

**Comment by feross** [comment]
@geastwood I haven't looked at this code in a long time. If you wouldn't mind, can you use comments to annotate why each line was changed? I super appreciate it!

**Comment by geastwood** [comment]
Just to bring it in line with nodejs on how to handle NaN and adding a test case for it.

**Comment by feross** [comment]
@geastwood Thanks, but I meant line by line. Like, what was wrong with each line, why did you change it?

**Comment by dcousens** [comment]
@geastwood why was _this_ implementation wrong,  not Node's?

**Review by Avatarebrsut — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+node_modules
```
Comment (Avatarebrsut) [inline .gitignore:1]: index.js

**Review by Avatarebrsut — COMMENTED** [review]
index.js

File: .gitignore (hunk 1 above)
Comment (Avatarebrsut) [reply .gitignore:1]: > index.js

**Review by Avatarebrsut — CHANGES_REQUESTED** [review]
index.js

**Review by Avatarebrsut — COMMENTED** [review]
gh pr checkout 31

**Pull request closed without merging**